pkg/abaputils/abaputils.go

```diff
@@ -0,0 +1,254 @@
+package abaputils
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// GetAbapCommunicationArrangementInfo function fetches the communcation arrangement information in SAP CP ABAP Environment
+func GetAbapCommunicationArrangementInfo(options AbapEnvironmentOptions, c command.ExecRunner, oDataURL string, cfLogoutOption bool) (ConnectionDetailsHTTP, error) {
+
+	var connectionDetails ConnectionDetailsHTTP
+	var error error
+
+	if options.Host != "" {
+		// Host, User and Password are directly provided -> check for host prefix (double https)
+		match, err := regexp.MatchString(`^[hH][tT][tT][pP][sS]:\/\/.*`, options.Host)
+		if err != nil {
+			return connectionDetails, errors.Wrap(err, "Host prefix validation failed")
+		}
+		if match {
+			connectionDetails.URL = options.Host + oDataURL
```

Comment:
[nit-pick] `options.Host + oDataURL` is the same in `if` and `else` branch. Following the pure doctrine this should be handle outside the `if`/`else` control flow.

Reply:
fixed